add neuron volume and taper analysis

carshadi opened the pull request:
add neuron volume and taper analysis

carshadi committed:
add neuron volume and taper analysis

add neuron volume and taper analysis

carshadi commented:
This pull request adds compartment volume analysis as well as mean burke taper vs. branch order

tferr commented:
Welcome to the SNT community @carshadi!
Thanks for your contribution. This is great!

A couple of comments:
- [ ] It would be important to add references to the formulas used, and which assumptions/approximations are made on volume calculation
- [ ] Also important is to compare the results against [L-measure](http://cng.gmu.edu:8080/Lm/) (or [btmorph](https://btmorph.readthedocs.io/en/latest/readme.html)?) to ensure the accuracy of the result. The simplest way would perhaps to load a neuron (with traced diameters) from http://neuromorpho.org/, since metrics of neuromorpho cells are already online
- [ ] Add some sanity checks warning users analyzing reconstructions without/(or constant) diameter
- [ ]  `ij.gui.plot` is the only ij1 legacy library that you are using. The IJ2 alternative is [PlotService](https://github.com/maarzt/imagej-plot-service). I am not sure if the API is already public, if it is, it would be nice to include it.
- [ ] Why not include Hillman taper? 
Do you agree? Do you need snippets for the above? Also, Is there anything else you would like to be reviewed?

Thanks once more for your contribution!

carshadi committed:
add necessary changes

added references to formulas and functions used, import an swc from NeuroMorpho.org and run an analysis on it. Changed the radii analysis to measure the mean path radius vs branch order instead of the taper. Rational is that plotting the radii is more generally informative. Tapering can be gleaned from taking the first derivative of a "node radius vs distance from soma" fitted curve at any arbitrary point. Also replaced the legacy plot with the new PlotService script parameter.

carshadi committed:
rename TreeColorizer to TreeColorMapper

tferr commented on src/main/resources/script_templates/Neuroanatomy/Analysis_Demo2.py:
```diff
@@ -16,11 +16,12 @@
 
 from tracing import (Path, PathAndFillManager, SimpleNeuriteTracer, SNTUI, Tree)
-from tracing.io import MLJSONLoader
+from tracing.io import MLJSONLoader, NeuroMorphoLoader
```
Several of these Imports have been renamed:
tracing.io.MJJSONLoader > tracing.io.MouseLightLoader
tracing.plot.TreePlot2D > tracing.viewer.Viewer2D
tracing.plot.TreePlot3D > tracing.viewer.Viewer3D
You can use the list in the script editor Templates>NeuroAnatomy>Boilerplate>Python here https://github.com/fiji/Simple_Neurite_Tracer/blob/scijava/src/main/resources/script_templates/Neuroanatomy/BoilerPlate/PY.py

carshadi replied:
imports have been renamed

tferr commented on src/main/resources/script_templates/Neuroanatomy/Analysis_Demo2.py:
```diff
@@ -6,7 +6,7 @@
+# @PlotService plotService
 
 '''
 file:       Analysis_Demo2.py
```
For completion, can you add your name to the author field? and a description of what the script does?

carshadi replied:
done

tferr reviewed (changes requested):
HI @carshadi, This is nice! I added some minor requests, so that it runs on the new API

carshadi reviewed (commented):
thanks @tferr , will update

carshadi commented:
I mistakenly created a new pull request reflecting the requested changes instead of updating to this one. New pull request here https://github.com/fiji/Simple_Neurite_Tracer/pull/42

tferr commented:
No problem! Thanks a lot! For future ref, this was superseded by #42 that meanwhile has been merged
